python/gps/algorithm/algorithm.py

```diff
@@ -23,7 +23,12 @@ def __init__(self, hyperparams):
         config.update(hyperparams)
         self._hyperparams = config
 
-        self.M = hyperparams['conditions']
+        if 'train_conditions' in hyperparams:
```

Comment:
This is a nice feature (having train and test conditions). Can you also mention it in the PR description?